### PR TITLE
Add jenkins file for manifest update workflow

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
This is a work around until #3068 gets resolved. Tested this jenkinsFile and it works as expected. Sample PR it opened https://github.com/opensearch-project/opensearch-build/pull/3428

### Issues Resolved
Related #3068

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
Let me know if I should remove this workflow https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/versions.yml

Thanks!

File: jenkins/manifests-update.jenkinsFile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            steps {
+                script {
+                    withCredentials([usernamePassword(credentialsId: 'jenkins-github-bot-token', passwordVariable: 'GITHUB_TOKEN', usernameVariable: 'GITHUB_USER')]) {
+                        sh """
+                        git remote set-url origin https://opensearch-ci:${GITHUB_TOKEN}@github.com/opensearch-project/opensearch-build
+                        git config user.email "opensearch-infra@amazon.com"
+                        git config user.name "opensearch-ci"
+                        git checkout -b update-manifest-${BUILD_NUMBER}
+                        ./manifests.sh update
+                        git add . && git commit -sm "Update manifests"
```
Comment (rishabh6788) [inline jenkins/manifests-update.jenkinsFile:30]: What happens when there are not changes to add or commit?

Comment (gaiksaya) [reply jenkins/manifests-update.jenkinsFile:30]: It throws an exception. Let me put a try catch block to handle that and also lessen the scope of github token

Comment (gaiksaya) [reply jenkins/manifests-update.jenkinsFile]: Added a try catch block that will handle the exception if anything goes wrong and exit gracefully if there is nothing to commit.
Thanks!

**Review by rishabh6788 — APPROVED** [review]

**Comment by gaiksaya** [comment]
Created a workflow with this jenkinsFile in CI system here https://build.ci.opensearch.org/job/manifest-update/

**Pull request merged**